File: tricktreat.cpp
#include <bits/stdc++.h>

using namespace std;

typedef pair<double,double> dd;
typedef vector<dd> vdd;

double distance(double x1, double y1, double x2, double y2) {
    return sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
}

double f(double x, double y, vdd houses) {
    double ans = -1;
    for (dd pair: houses) {
        ans = max(ans, distance(pair.first,pair.second,x,y));
    }
    return ans;
}

int main() {
    double EPSILON = 0.00001;
    while (true) {
        int n; scanf("%d",&n);
        if (n == 0) break;
        vdd houses;
        for (int i = 0; i < n; i++) {
            double x,y; scanf("%lf %lf",&x,&y);
            houses.push_back(make_pair(x,y));
        }
        //perform ternary search
        double L = -200000, H = 200000;
        while (H - L > EPSILON) {
            double d = (H - L) / 3.0;
            double m1 = L + d, m2 = H - d;
            (f(m1,0,houses) > f(m2,0,houses)) ? L = m1 : H = m2;
        }
        double ans = (H + L) / 2.0;
        printf("%lf %lf\n",ans,f(ans,0,houses));
    }
    return 0;
}

File: tricktreat.py
from sys import stdin,stdout
from math import sqrt

def dist(x1,y1,x2,y2):
    return sqrt((x2 - x1)**2 + (y2 - y1)**2)

#f is decreasing and increasing function 
def f(x,y,houses):
    return max(dist(houses[i][0],houses[i][1],x,y) for i in range(len(houses)))

EPSILON = 10**(-5)
while True:
    n = int(stdin.readline())
    if n == 0:
        break
    houses = [tuple(map(float,stdin.readline().split())) for _ in range(n)]
    L,H = -200000,200000
    while H - L > EPSILON: #perform ternary search
        d = (H - L) / 3
        m1,m2 = L + d, H - d
        if f(m1,0,houses) > f(m2,0,houses):
            L = m1
        else:
            H = m2
    ans = (L + H) / 2
    stdout.write(f"{ans:.7f} {f(ans,0,houses):.7f}\n") 
    stdin.readline() #read in blank line
